documentation/helpers/ToC/index.jsx

```diff
@@ -0,0 +1,68 @@
+import { cva, cx } from 'class-variance-authority'
+import { useEffect, useState } from 'react'
+
+import { useActiveAnchor } from './useActiveAnchor'
+import { scrollToAnchor, updateUrlAnchor } from './utils'
+
+const itemStyle = cva(['block', 'py-sm', ['hover:bg-primary-container-hovered']], {
+  variants: {
+    active: {
+      true: ['text-primary', 'shadow-[inset_2px_0px_0px]'],
+      false: ['text-on-surface', 'shadow-[inset_1px_0px_0px_#D1D0D5]'],
+    },
+    tagName: {
+      H2: 'pl-md text-body-1',
+      H3: "pl-[calc(theme('spacing.xl')*1)] text-body-2",
+      H4: "pl-[calc(theme('spacing.xl')*2)] text-caption",
+    },
+  },
+})
+
+export const ToC = () => {
+  const [headings, setHeadings] = useState(null)
+
+  useEffect(() => {
+    const list = document.querySelectorAll('h2, h3')
+
+    if ([...list].length) {
+      setHeadings(document.querySelectorAll('h2, h3'))
+    }
+  }, [])
+
+  console.log(headings)
+  const activeAnchor = useActiveAnchor(headings)
+
+  return headings ? (
```

Comment:
```suggestion
  if (!headings.length) return
  return headings.map(...)
```
or
```suggestion
  if (!headings) return
  return headings.map(...)
```

Instead of using an expanded ternary expression (which requires scanning through multiple lines of code to get to the other side of the ternary), maybe we could use an early return instead